python/itools-y4m.py

```diff
@@ -199,7 +185,20 @@ def read_frame(self, data, output_colorrange, debug):
         elif self.colorspace in ("444",):
             chroma_w = self.width
             chroma_h = self.height
+        elif self.colorspace in ("420p10",):
+            chroma_w = self.width >> 1
+            chroma_h = self.height >> 1
+            luma_size *= 2
+            dt = np.dtype(np.uint16)
+
+        ya = np.frombuffer(data[offset : offset + luma_size], dtype=dt).reshape(
```

Comment:
Now here you're reading the luminance

Reply:
Please move it after you've done all the calculations (before line 202 in your patch)